runtime/common/src/precompile/honzon.rs

```diff
@@ -0,0 +1,523 @@
+				})
+			}
+			Action::CloseLoanByDex => {
+				let who = input.account_id_at(1)?;
+				let currency_id = input.currency_id_at(2)?;
+				let max_collateral_amount = input.balance_at(3)?;
+
+				log::debug!(
+					target: "evm",
+					"honzon: close_loan_by_dex who: {:?}, currency_id: {:?}, max_collateral_adjustment: {:?}",
+					who, currency_id, max_collateral_amount
+				);
+
+				<module_honzon::Pallet<Runtime> as HonzonManager<Runtime::AccountId, CurrencyId, Amount, Balance>>::close_loan_by_dex(who, currency_id, max_collateral_amount).map_err(|e|
+                    PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: Into::<&str>::into(e).as_bytes().to_vec(),
+                        cost: target_gas_limit(target_gas).unwrap_or_default(),
+                    }
+                )?;
+
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: vec![],
+					logs: Default::default(),
+				})
+			}
+			Action::GetPosition => {
+				let who = input.account_id_at(1)?;
+				let currency_id = input.currency_id_at(2)?;
+
+				let Position { collateral, debit } = <module_honzon::Pallet<Runtime> as HonzonManager<
+					Runtime::AccountId,
+					CurrencyId,
+					Amount,
+					Balance,
+				>>::get_position(&who, currency_id);
+
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: Output::default().encode_u128_tuple(collateral, debit),
+					logs: Default::default(),
+				})
+			}
+			Action::GetLiquidationRatio => {
+				let currency_id = input.currency_id_at(1)?;
+				let ratio = <module_honzon::Pallet<Runtime> as HonzonManager<
+					Runtime::AccountId,
+					CurrencyId,
+					Amount,
+					Balance,
+				>>::get_liquidation_ratio(currency_id)
+				.ok_or(PrecompileFailure::Revert {
+					exit_status: ExitRevert::Reverted,
+					output: "Honzon get liquidation ratio failed".into(),
+					cost: target_gas_limit(target_gas).unwrap_or_default(),
+				})?;
+
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: Output::default().encode_u128(ratio.into_inner()),
+					logs: Default::default(),
+				})
+			}
+			Action::GetCurrentCollateralRatio => {
+				let who = input.account_id_at(1)?;
+				let currency_id = input.currency_id_at(2)?;
+				let ratio = <module_honzon::Pallet<Runtime> as HonzonManager<
+					Runtime::AccountId,
+					CurrencyId,
+					Amount,
+					Balance,
+				>>::get_current_collateral_ratio(&who, currency_id)
+				.ok_or(PrecompileFailure::Revert {
+					exit_status: ExitRevert::Reverted,
+					output: "Honzon get current collateral ratio failed, no price feed".into(),
+					cost: target_gas_limit(target_gas).unwrap_or_default(),
+				})?;
+
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: Output::default().encode_u128(ratio.into_inner()),
```

Comment:
@ermalkaleci should we use encode_u256 to match the `return abi.decode(returnData, (uint256));` ?

Reply:
@zjb0807 I think this is fine because an encoded u128 can be decoded into u256 but I think we should check actions so input that is required to be u128 isn't passed as u256